src/main/resources/language/messages_pl.properties

```diff
@@ -58,18 +58,25 @@ nonstandard.checkBalanceButton = Sprawd\u017A Bilans
 logout.message = Zosta\u0142e\u015B prawid\u0142owo wylogowany!
 
 income.form = Przych\u00F3d
-income.form.amount = Kwota
-income.form.date = Data
-income.form.category = Kategoria
-income.form.category.choice = -- Prosz\u0119 wybra\u0107 kategori\u0119 --
-income.form.comment = Komentarz
-income.form.add = Dodaj
-income.form.cancel = Cofnij
-
-income.amount.empty = Kwota jest wymagana
-income.date.empty = Data jest wymagana
-income.category.empty = Kategoria jest wymagana
-income.success = Przych\u00F3d zosta\u0142 dodany
+expense.form = Wydatek
+
+form.amount = Kwota
+form.date = Data
+form.payment = Spos\u00F3b p\u0142atno\u015Bci
+form.category = Kategoria
+form.category.choice = -- Prosz\u0119 wybra\u0107 kategori\u0119 --
+form.payment.choice = -- Prosz\u0119 wybra\u0107 spos\u00F3b p\u0142atno\u015Bci --
+form.comment = Komentarz
+form.add = Dodaj
+form.cancel = Cofnij
+
+amount.empty = Kwota jest wymagana
+date.empty = Data jest wymagana
+payment.empty = Spos\u00F3b p\u0142atno\u015Bci jest wymagany
+category.empty = Kategoria jest wymagana
```

Comment:
The translation keys should be defined in such a way that without an element where it is used we know the content of the translation key. The keys need to be very specific. If those are errors, or validation messages it should be in the name. 

so for example: expenseForm.expenseCategory.error.empty
Don't try to create a reusable translation keys - from my experience it causes more problems than it simplifies. 
So if there are two forms, create two same lists of translations.
Let's say you use the same translation for "category" for two forms. Your product owner comes and wants to make it more precise and asks you to change it to "expense category". You may very quickly forget it is used in the other form as well and both icome and expense form will get "expense category". My advice is to never fear having too many translations ;)

Reply:
Explained very well ;)